meeshkan/serve/mock/request_processor.py

```diff
@@ -0,0 +1,100 @@
+import json
+import logging
+import random
+from typing import Sequence
+
+from http_types import Request, Response
+
+from meeshkan.serve.mock.callbacks import CallbackManager
+from meeshkan.serve.mock.faker.faker_exception import FakerException
+from meeshkan.serve.mock.faker.stateful_faker import StatefulFaker
+from meeshkan.serve.mock.matcher import match_request_to_openapi
+from meeshkan.serve.mock.rest import RestMiddlewareManager
+from meeshkan.serve.mock.security import match_to_security_schemes
+from meeshkan.serve.mock.specs import OpenAPISpecification
+from meeshkan.serve.mock.storage.mock_data_store import MockDataStore
+
+logger = logging.getLogger(__name__)
+
+
+class RequestProcessor:
+    _specs: Sequence[OpenAPISpecification]
+
+    def __init__(
+        self,
+        specs: Sequence[OpenAPISpecification],
+        mock_data_store: MockDataStore,
+        callback_manager: CallbackManager,
+        rest_middleware_manager: RestMiddlewareManager,
+    ):
+        self._specs = specs
+        self._mock_data_store = mock_data_store
+        self._callback_manager = callback_manager
+        self._rest_middleware_manager = rest_middleware_manager
+        self._faker = StatefulFaker(self._mock_data_store)
+
+        for spec in specs:
+            self._mock_data_store.add_mock(spec.source, spec.api)
```

Comment:
I think this is still a surprising side-effect of initializing the `RequestProcessor` but no need to change it!